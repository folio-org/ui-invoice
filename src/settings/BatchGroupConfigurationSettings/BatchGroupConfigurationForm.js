import React, { useCallback } from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import { Field } from 'react-final-form';
import { OnChange } from 'react-final-form-listeners';
import { get } from 'lodash';

import stripesFinalForm from '@folio/stripes/final-form';
import {
  Button,
  Col,
  ConfirmationModal,
  Pane,
  Row,
  TextField,
  Timepicker,
} from '@folio/stripes/components';
import {
  FieldSelectFinal as FieldSelect,
  useModalToggle,
  validateRequired,
} from '@folio/stripes-acq-components';

import BatchGroupConfigurationFormFooter from './BatchGroupConfigurationFormFooter';
import BatchGroupsField from './BatchGroupsField';
import TogglePassword from './TogglePassword';
import WeekdaysField from './WeekdaysField';
import { validateUploadURI } from './utils';
import {
  EXPORT_FORMAT_OPTIONS,
  SCHEDULE_EXPORT_OPTIONS,
  SCHEDULE_EXPORT,
  WEEKDAYS_OPTIONS,
} from './constants';
import { BatchVoucherExportsList } from './BatchVoucherExportsList';

const trimTime = value => value.slice(0, 5);

const BatchGroupConfigurationForm = ({
  batchGroups,
  batchVoucherExports,
  form,
  handleSubmit,
  hasCredsSaved,
  onNeedMoreData,
  pristine,
  recordsCount,
  runManualExport,
  selectBatchGroup,
  selectedBatchGroupId,
  submitting,
  testConnection,
}) => {
  const formValues = get(form.getState(), 'values', {});
  const scheduleExportWeekly = formValues.scheduleExport === SCHEDULE_EXPORT.weekly;
  const [isManualExportConfirmation, toggleManualExportConfirmation] = useModalToggle();
  const selectedBatchGroupName = batchGroups.find(({ id }) => id === selectedBatchGroupId)?.name;

  const onRunManualExport = useCallback(
    () => {
      toggleManualExportConfirmation();
      runManualExport();
    },
    [runManualExport, toggleManualExportConfirmation],
  );

  const paneFooter = (
    <BatchGroupConfigurationFormFooter
      exportConfigId={formValues.id}
      handleSubmit={handleSubmit}
      pristine={pristine}
      runManualExport={toggleManualExportConfirmation}
      submitting={submitting}
    />
  );

  return (
    <Pane
      data-test-batch-group-configuration-settings
      defaultWidth="fill"
      footer={paneFooter}
      id="pane-batch-group-configuration"
      paneTitle={<FormattedMessage id="ui-invoice.settings.batchGroupConfiguration.label" />}
    >
      <Row>
        <Col
          data-test-col-batch-group-field
          xs={4}
        >
          <BatchGroupsField
            batchGroups={batchGroups}
            selectBatchGroup={selectBatchGroup}
            selectedBatchGroupId={selectedBatchGroupId}
          />
        </Col>
      </Row>

      <Row>
        <Col
          data-test-col-schedule-export
          xs={4}
        >
          <FieldSelect
            dataOptions={SCHEDULE_EXPORT_OPTIONS}
            label={<FormattedMessage id="ui-invoice.settings.batchGroupConfiguration.scheduleExport" />}
            name="scheduleExport"
          />
          <OnChange name="scheduleExport">
            {form.mutators.setScheduledExport}
          </OnChange>
        </Col>
      </Row>

      {formValues.enableScheduledExport && (
        <>
          {scheduleExportWeekly && (
            <Row>
              <Col
                data-test-col-weekdays
                xs={12}
              >
                <WeekdaysField
                  name="weekdays"
                  weekdays={WEEKDAYS_OPTIONS}
                />
              </Col>
            </Row>
          )}

          <Row>
            <Col
              data-test-col-time
              xs={4}
            >
              <Field
                name="startTime"
                component={Timepicker}
                label={<FormattedMessage id="ui-invoice.settings.batchGroupConfiguration.setTime" />}
                parse={trimTime}
                required
                validate={validateRequired}
              />
            </Col>
          </Row>
        </>
      )}

      <Row>
        <Col
          data-test-col-upload-location
          xs={8}
        >
          <Field
            component={TextField}
            id="uploadURI"
            label={<FormattedMessage id="ui-invoice.settings.batchGroupConfiguration.uploadLocation" />}
            name="uploadURI"
            type="text"
            validate={validateUploadURI}
          />
        </Col>
        <Col
          data-test-col-format
          xs={4}
        >
          <FieldSelect
            dataOptions={EXPORT_FORMAT_OPTIONS}
            label={<FormattedMessage id="ui-invoice.settings.batchGroupConfiguration.format" />}
            name="format"
            required
            validate={validateRequired}
          />
        </Col>
      </Row>

      <Row bottom="xs">
        <Col
          data-test-col-username
          xs={4}
        >
          <Field
            id="username"
            label={<FormattedMessage id="ui-invoice.settings.batchGroupConfiguration.username" />}
            name="username"
            component={TextField}
            fullWidth
            parse={v => v}
          />
        </Col>

        <TogglePassword name="password" />
      </Row>

      <Row>
        <Col
          data-test-col-test-connection
          xs={4}
        >
          <Button
            buttonStyle="primary"
            bottomMargin0
            data-test-connection-test-button
            disabled={!(formValues.id && formValues.uploadURI && hasCredsSaved)}
            onClick={testConnection}
          >
            <FormattedMessage id="ui-invoice.settings.batchGroupConfiguration.testConnection" />
          </Button>
        </Col>
      </Row>
      <BatchVoucherExportsList
        batchVoucherExports={batchVoucherExports}
        format={formValues.format}
        onNeedMoreData={onNeedMoreData}
        recordsCount={recordsCount}
      />

      {isManualExportConfirmation && (
        <ConfirmationModal
          id="run-manual-export-confirmation"
          confirmLabel={<FormattedMessage id="ui-invoice.button.continue" />}
          heading={<FormattedMessage id="ui-invoice.settings.actions.manualExport.heading" />}
          message={
            <FormattedMessage
              id="ui-invoice.settings.actions.manualExport.message"
              values={{ selectedBatchGroupName }}
            />
          }
          onCancel={toggleManualExportConfirmation}
          onConfirm={onRunManualExport}
          open
        />
      )}
    </Pane>
  );
};

BatchGroupConfigurationForm.propTypes = {
  batchGroups: PropTypes.arrayOf(PropTypes.object),
  batchVoucherExports: PropTypes.arrayOf(PropTypes.object),
  form: PropTypes.object,
  handleSubmit: PropTypes.func.isRequired,
  hasCredsSaved: PropTypes.bool.isRequired,
  onNeedMoreData: PropTypes.func,
  pristine: PropTypes.bool,
  recordsCount: PropTypes.number,
  runManualExport: PropTypes.func.isRequired,
  selectBatchGroup: PropTypes.func.isRequired,
  selectedBatchGroupId: PropTypes.string,
  submitting: PropTypes.bool,
  testConnection: PropTypes.func.isRequired,
};

export default stripesFinalForm({
  subscription: { values: true },
  navigationCheck: true,
  mutators: {
    setScheduledExport: ([value], state, tools) => {
      if (!value) {
        tools.changeValue(state, 'enableScheduledExport', () => false);
        tools.changeValue(state, 'startTime', () => null);
        tools.changeValue(state, 'weekdays', () => {});
      } else if (value === SCHEDULE_EXPORT.daily) {
        tools.changeValue(state, 'enableScheduledExport', () => true);
        tools.changeValue(state, 'weekdays', () => {});
      } else {
        tools.changeValue(state, 'enableScheduledExport', () => true);
      }
    },
  },
})(BatchGroupConfigurationForm);
