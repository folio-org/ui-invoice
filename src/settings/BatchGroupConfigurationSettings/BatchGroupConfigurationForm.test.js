import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';

import { batchGroup, batchVoucherExport } from '../../../test/jest/fixtures';
import {
  SCHEDULE_EXPORT,
} from './constants';
import BatchGroupConfigurationForm from './BatchGroupConfigurationForm';

const defaultProps = {
  batchGroups: [batchGroup],
  batchVoucherExports: [batchVoucherExport],
  hasCredsSaved: true,

  initialValues: {
    id: 'exportId',
    enableScheduledExport: true,
    uploadURI: 'https://folio.com',
    scheduleExport: SCHEDULE_EXPORT.weekly,
  },
  onSubmit: jest.fn(),

  testConnection: jest.fn(),
  selectBatchGroup: jest.fn(),
};
const renderBatchGroupConfigurationForm = (props = defaultProps) => render(
  <BatchGroupConfigurationForm {...props} />,
  { wrapper: MemoryRouter },
);

describe('BatchGroupConfigurationForm component', () => {
  beforeEach(() => {
    global.document.createRange = global.document.originalCreateRange;
  });

  afterEach(() => {
    global.document.createRange = global.document.mockCreateRange;
  });

  xit('should render correct structure', () => {
    const { asFragment } = renderBatchGroupConfigurationForm();

    expect(asFragment()).toMatchSnapshot();
  });
});
