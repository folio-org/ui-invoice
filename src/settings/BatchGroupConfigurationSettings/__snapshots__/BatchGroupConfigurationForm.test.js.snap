// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BatchGroupConfigurationForm component should render correct structure 1`] = `
<DocumentFragment>
  <section
    class="pane focusIndicator"
    data-test-batch-group-configuration-settings="true"
    id="pane-batch-group-configuration"
    tabindex="-1"
  >
    <div
      aria-labelledby="paneHeaderpane-batch-group-configuration-pane-title"
      class="paneHeader"
      data-test-pane-header="true"
      id="paneHeaderpane-batch-group-configuration"
      tabindex="-1"
    >
      <div
        class="paneHeaderCenter"
      >
        <div
          class="paneHeaderCenterInner"
          id="paneHeaderpane-batch-group-configuration-pane-title"
        >
          <h2
            class="paneTitle"
            data-test-pane-header-title="true"
          >
            <span
              class="paneTitleLabel"
            >
              ui-invoice.settings.batchGroupConfiguration.label
            </span>
          </h2>
        </div>
      </div>
    </div>
    <div
      class="paneContent hasPadding"
      id="pane-batch-group-configuration-content"
      tabindex="-1"
    >
      <div
        class="row"
      >
        <div
          class="col-xs-4"
          data-test-col-batch-group-field="true"
        >
          <div
            class="kvRoot"
            data-test-batch-group-value="true"
          >
            <div
              class="kvLabel"
            >
              ui-invoice.settings.batchGroupConfiguration.batchGroup
            </div>
            <div
              class="kvValue"
              data-test-kv-value="true"
            >
              Amherst College (AC)
            </div>
          </div>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-xs-4"
          data-test-col-schedule-export="true"
        >
          <div
            class="select inputGroup fullWidth"
          >
            <label
              class="label"
              for="select-2"
              id="select-2-label"
            >
              ui-invoice.settings.batchGroupConfiguration.scheduleExport
            </label>
            <div
              class="selectWrap"
            >
              <select
                aria-invalid="false"
                aria-labelledby="select-2-label"
                aria-required="false"
                class="selectControl formControl"
                id="select-2"
                name="scheduleExport"
              >
                <option
                  value=""
                >
                  stripes-acq-components.label.emptyValue
                </option>
                <option
                  value="daily"
                >
                  ui-invoice.settings.batchGroupConfiguration.scheduleExport.daily
                </option>
                <option
                  value="weekly"
                >
                  ui-invoice.settings.batchGroupConfiguration.scheduleExport.weekly
                </option>
              </select>
              <span>
                Icon
              </span>
            </div>
            <div
              role="alert"
            />
          </div>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-xs-12"
          data-test-col-weekdays="true"
        >
          <label
            class="label"
          >
            ui-invoice.settings.batchGroupConfiguration.on
            <span
              aria-hidden="true"
              class="asterisk"
            >
              *
            </span>
          </label>
          <div
            class="checkbox inline vertical"
            data-test-checkbox="true"
          >
            <label
              class="label label checkboxInteractionStylesControl"
              for="checkbox-12"
            >
              <span
                class="labelText"
              >
                ui-invoice.settings.batchGroupConfiguration.weekdays.Sunday
              </span>
              <span
                class="inner checkboxInteractionStyles"
              >
                <input
                  aria-invalid="false"
                  class="input"
                  id="checkbox-12"
                  name="weekdays[Sunday]"
                  type="checkbox"
                  value=""
                />
                <span
                  class="checkboxIcon"
                >
                  <svg
                    height="10"
                    viewBox="0 0 14 14"
                    width="10"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M10.395 4.617L5.63 9.383 3.605 7.358"
                      fill="none"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-miterlimit="10"
                      stroke-width="1.5"
                    />
                  </svg>
                </span>
              </span>
            </label>
            <div
              role="alert"
            />
          </div>
          <div
            class="checkbox inline vertical"
            data-test-checkbox="true"
          >
            <label
              class="label label checkboxInteractionStylesControl"
              for="checkbox-13"
            >
              <span
                class="labelText"
              >
                ui-invoice.settings.batchGroupConfiguration.weekdays.Monday
              </span>
              <span
                class="inner checkboxInteractionStyles"
              >
                <input
                  aria-invalid="false"
                  class="input"
                  id="checkbox-13"
                  name="weekdays[Monday]"
                  type="checkbox"
                  value=""
                />
                <span
                  class="checkboxIcon"
                >
                  <svg
                    height="10"
                    viewBox="0 0 14 14"
                    width="10"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M10.395 4.617L5.63 9.383 3.605 7.358"
                      fill="none"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-miterlimit="10"
                      stroke-width="1.5"
                    />
                  </svg>
                </span>
              </span>
            </label>
            <div
              role="alert"
            />
          </div>
          <div
            class="checkbox inline vertical"
            data-test-checkbox="true"
          >
            <label
              class="label label checkboxInteractionStylesControl"
              for="checkbox-14"
            >
              <span
                class="labelText"
              >
                ui-invoice.settings.batchGroupConfiguration.weekdays.Tuesday
              </span>
              <span
                class="inner checkboxInteractionStyles"
              >
                <input
                  aria-invalid="false"
                  class="input"
                  id="checkbox-14"
                  name="weekdays[Tuesday]"
                  type="checkbox"
                  value=""
                />
                <span
                  class="checkboxIcon"
                >
                  <svg
                    height="10"
                    viewBox="0 0 14 14"
                    width="10"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M10.395 4.617L5.63 9.383 3.605 7.358"
                      fill="none"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-miterlimit="10"
                      stroke-width="1.5"
                    />
                  </svg>
                </span>
              </span>
            </label>
            <div
              role="alert"
            />
          </div>
          <div
            class="checkbox inline vertical"
            data-test-checkbox="true"
          >
            <label
              class="label label checkboxInteractionStylesControl"
              for="checkbox-15"
            >
              <span
                class="labelText"
              >
                ui-invoice.settings.batchGroupConfiguration.weekdays.Wednesday
              </span>
              <span
                class="inner checkboxInteractionStyles"
              >
                <input
                  aria-invalid="false"
                  class="input"
                  id="checkbox-15"
                  name="weekdays[Wednesday]"
                  type="checkbox"
                  value=""
                />
                <span
                  class="checkboxIcon"
                >
                  <svg
                    height="10"
                    viewBox="0 0 14 14"
                    width="10"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M10.395 4.617L5.63 9.383 3.605 7.358"
                      fill="none"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-miterlimit="10"
                      stroke-width="1.5"
                    />
                  </svg>
                </span>
              </span>
            </label>
            <div
              role="alert"
            />
          </div>
          <div
            class="checkbox inline vertical"
            data-test-checkbox="true"
          >
            <label
              class="label label checkboxInteractionStylesControl"
              for="checkbox-16"
            >
              <span
                class="labelText"
              >
                ui-invoice.settings.batchGroupConfiguration.weekdays.Thursday
              </span>
              <span
                class="inner checkboxInteractionStyles"
              >
                <input
                  aria-invalid="false"
                  class="input"
                  id="checkbox-16"
                  name="weekdays[Thursday]"
                  type="checkbox"
                  value=""
                />
                <span
                  class="checkboxIcon"
                >
                  <svg
                    height="10"
                    viewBox="0 0 14 14"
                    width="10"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M10.395 4.617L5.63 9.383 3.605 7.358"
                      fill="none"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-miterlimit="10"
                      stroke-width="1.5"
                    />
                  </svg>
                </span>
              </span>
            </label>
            <div
              role="alert"
            />
          </div>
          <div
            class="checkbox inline vertical"
            data-test-checkbox="true"
          >
            <label
              class="label label checkboxInteractionStylesControl"
              for="checkbox-17"
            >
              <span
                class="labelText"
              >
                ui-invoice.settings.batchGroupConfiguration.weekdays.Friday
              </span>
              <span
                class="inner checkboxInteractionStyles"
              >
                <input
                  aria-invalid="false"
                  class="input"
                  id="checkbox-17"
                  name="weekdays[Friday]"
                  type="checkbox"
                  value=""
                />
                <span
                  class="checkboxIcon"
                >
                  <svg
                    height="10"
                    viewBox="0 0 14 14"
                    width="10"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M10.395 4.617L5.63 9.383 3.605 7.358"
                      fill="none"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-miterlimit="10"
                      stroke-width="1.5"
                    />
                  </svg>
                </span>
              </span>
            </label>
            <div
              role="alert"
            />
          </div>
          <div
            class="checkbox inline vertical"
            data-test-checkbox="true"
          >
            <label
              class="label label checkboxInteractionStylesControl"
              for="checkbox-18"
            >
              <span
                class="labelText"
              >
                ui-invoice.settings.batchGroupConfiguration.weekdays.Saturday
              </span>
              <span
                class="inner checkboxInteractionStyles"
              >
                <input
                  aria-invalid="false"
                  class="input"
                  id="checkbox-18"
                  name="weekdays[Saturday]"
                  type="checkbox"
                  value=""
                />
                <span
                  class="checkboxIcon"
                >
                  <svg
                    height="10"
                    viewBox="0 0 14 14"
                    width="10"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M10.395 4.617L5.63 9.383 3.605 7.358"
                      fill="none"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-miterlimit="10"
                      stroke-width="1.5"
                    />
                  </svg>
                </span>
              </span>
            </label>
            <div
              role="alert"
            />
          </div>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-xs-4"
          data-test-col-time="true"
        >
          <div
            style="position: relative; width: 100%;"
          >
            <div
              aria-live="assertive"
              aria-relevant="additions"
              class="sr-only"
            >
              <div />
            </div>
            <div
              class="textField"
            >
              <label
                class="label"
                for="timepick-10"
              >
                ui-invoice.settings.batchGroupConfiguration.setTime
                <span
                  aria-hidden="true"
                  class="asterisk"
                >
                  *
                </span>
              </label>
              <div
                class="formControl inputGroup"
              >
                <input
                  aria-invalid="false"
                  aria-label="[object Object] format is h:m m  A use arrow keys to navigate dates. Enter key to confirm. "
                  aria-required="true"
                  autocapitalize="off"
                  autocomplete="off"
                  autocorrect="off"
                  data-test-timepicker-input="true"
                  error=""
                  id="timepick-10"
                  name="startTime"
                  placeholder=" h:mm A"
                  required=""
                  spellcheck="false"
                  type="text"
                  value=""
                  warning=""
                />
                <div
                  class="endControls"
                >
                  <div
                    class="controlGroup"
                  >
                    <button
                      aria-label="stripes-components.showOrHideTimepicker"
                      class="iconButton medium"
                      icon="clock"
                      id="timepicker-toggle-button-timepick-10"
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        class="iconButtonInner mediumOffset"
                      >
                        <span>
                          Icon
                        </span>
                      </span>
                    </button>
                  </div>
                </div>
              </div>
              <div
                role="alert"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-xs-8"
          data-test-col-upload-location="true"
        >
          <div
            class="textField"
          >
            <label
              class="label"
              for="uploadURI"
            >
              ui-invoice.settings.batchGroupConfiguration.uploadLocation
            </label>
            <div
              class="formControl inputGroup"
            >
              <input
                aria-invalid="false"
                autocapitalize="off"
                autocomplete="off"
                autocorrect="off"
                error=""
                id="uploadURI"
                name="uploadURI"
                spellcheck="false"
                type="text"
                value="https://folio.com"
                warning=""
              />
            </div>
            <div
              role="alert"
            />
          </div>
        </div>
        <div
          class="col-xs-4"
          data-test-col-format="true"
        >
          <div
            class="select inputGroup fullWidth"
          >
            <label
              class="label"
              for="select-11"
              id="select-11-label"
            >
              ui-invoice.settings.batchGroupConfiguration.format
              <span
                aria-hidden="true"
                class="asterisk"
              >
                *
              </span>
            </label>
            <div
              class="selectWrap"
            >
              <select
                aria-invalid="false"
                aria-labelledby="select-11-label"
                aria-required="true"
                class="selectControl formControl"
                id="select-11"
                name="format"
                required=""
              >
                <option
                  value=""
                >
                  stripes-acq-components.label.emptyValue
                </option>
                <option
                  value="Application/json"
                >
                  ui-invoice.settings.batchGroupConfiguration.format.Application/json
                </option>
                <option
                  value="Application/xml"
                >
                  ui-invoice.settings.batchGroupConfiguration.format.Application/xml
                </option>
              </select>
              <span>
                Icon
              </span>
            </div>
            <div
              role="alert"
            />
          </div>
        </div>
      </div>
      <div
        class="row bottom-xs"
      >
        <div
          class="col-xs-4"
          data-test-col-username="true"
        >
          <div
            class="textField"
          >
            <label
              class="label"
              for="username"
            >
              ui-invoice.settings.batchGroupConfiguration.username
            </label>
            <div
              class="formControl inputGroup"
            >
              <input
                aria-invalid="false"
                autocapitalize="off"
                autocomplete="off"
                autocorrect="off"
                error=""
                id="username"
                name="username"
                spellcheck="false"
                type="text"
                value=""
                warning=""
              />
            </div>
            <div
              role="alert"
            />
          </div>
        </div>
        <div
          class="col-xs-4"
        >
          <div
            class="textField"
          >
            <label
              class="label"
              for="password"
            >
              ui-invoice.settings.batchGroupConfiguration.password
            </label>
            <div
              class="formControl inputGroup"
            >
              <input
                aria-invalid="false"
                autocapitalize="off"
                autocomplete="new-password"
                autocorrect="off"
                error=""
                id="password"
                name="password"
                spellcheck="false"
                type="password"
                value=""
                warning=""
              />
            </div>
            <div
              role="alert"
            />
          </div>
        </div>
        <div
          class="col-xs-2"
        >
          <button
            class="button default"
            data-test-toggle-password-button="true"
            style="margin-bottom: 15px;"
            type="button"
          >
            <span
              class="inner"
            >
              ui-invoice.settings.batchGroupConfiguration.password.show
            </span>
          </button>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-xs-4"
          data-test-col-test-connection="true"
        >
          <button
            class="button primary marginBottom0"
            data-test-connection-test-button="true"
            type="button"
          >
            <span
              class="inner"
            >
              ui-invoice.settings.batchGroupConfiguration.testConnection
            </span>
          </button>
        </div>
      </div>
      <div
        style="overflow: visible; height: 0px; width: 0px;"
      />
      <div
        class="resize-triggers"
      >
        <div
          class="expand-trigger"
        >
          <div
            style="width: 1px; height: 1px;"
          />
        </div>
        <div
          class="contract-trigger"
        />
      </div>
    </div>
    <div
      class="paneFooter"
    >
      <div
        class="paneFooterContent"
      >
        <span
          class="renderStart"
          data-test-pane-footer-start="true"
        >
          <button
            class="button default mega"
            data-test-run-manual-export-button="true"
            type="button"
          >
            <span
              class="inner"
            >
              ui-invoice.button.runManualExport
            </span>
          </button>
        </span>
        <span
          data-test-pane-footer-end="true"
        >
          <button
            class="button primary mega"
            data-test-save-button="true"
            disabled=""
            type="submit"
          >
            <span
              class="inner"
            >
              ui-invoice.button.save
            </span>
          </button>
        </span>
      </div>
    </div>
  </section>
  <div />
</DocumentFragment>
`;
